internal/cmd/gencli/input.go

```diff
@@ -0,0 +1,339 @@
+package main
+
+import (
+	"github.com/hashicorp/boundary/internal/types/resource"
+)
+
+type cmdInfo struct {
+	// The type of the resource, e.g. "target"
+	ResourceType string
+
+	// The API package under the api path
+	Pkg string
+
+	// Standard actions (with standard parameters) used by this resource
+	StdActions []string
+
+	// HasExtraCommandVars controls whether to generate an embedded struct with
+	// extra command variables
+	HasExtraCommandVars bool
+
+	// SkipNormalHelp indicates skipping the normal help case for when it needs
+	// to be only custom (mainly for subactions)
+	SkipNormalHelp bool
+
+	// HasExtraHelpFunc controls whether to include a default statement chaining
+	// to an extra help function
+	HasExtraHelpFunc bool
+
+	// HasExampleCliOutput controls whether to generate code to look for a CLI
+	// output env var and print
+	HasExampleCliOutput bool
+
+	// IsAbstractType triggers some behavior specialized for abstract types,
+	// e.g. those that have subcommands for create/update
+	IsAbstractType bool
+
+	// HasId controls whether to add ID emptiness checking. Note that some
+	// commands that allow name/scope id or name/scope name handling may skip
+	// this in favor of custom logic.
+	HasId bool
+
+	// Container controls what to generate for a required container (scope ID,
+	// auth method ID, etc.); see ContainerRequiredActions as well
+	Container string
+
+	// ContainerRequiredActions controls which actions require that the
+	// container ID is not empty.
+	ContainerRequiredActions []string
+
+	// HasName controls whether to add name options
+	HasName bool
+
+	// HasDescription controls whether to add description options
+	HasDescription bool
+
+	// HasScopeName controls whether to add scope name options
+	HasScopeName bool
```

Comment:
I figured I'd keep this at the upper level because over time I think we'll be enabling more resources to be described by scope name/resource name than just targets.  But if you feel strongly about it I can move it directly into the targets extra funcs.